microsoft/terminal
# Check spelling 0.0.22

## Summary of the Pull Request

Upgrades [check-spelling to v0.0.22](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.22)

## References and Relevant Issues

## Detailed Description of the Pull Request / Additional comments

* refreshes workflow
  * enables dependabot PRs to trigger CI (so that in the future you'll be able to see breaking changes to the dictionary paths)
* refreshes metadata
* built-in handling of `\n`/`\r`/`\t` is removed -- This means that the `patterns/0_*.txt` files can be removed.
* this specific PR includes some shim content, in `allow/check-spelling-0.0.21.txt` -- once it this PR merges, it can be removed on a branch and the next CI will clean out items from `expect.txt` relating to the `\r` stuff and suggest replacement content.
* talking to the bot is enabled for forks (but not the master repository)
* SARIF reporting is enabled for PRs w/in a single repository (not across forks)
* In job reports, there's a summary table (space permitting) linking to instances (this is a poor man's SARIF report)
* When a pattern splits a thing that results in check-spelling finding an unrecognized token, that's reported with a distinct category
* When there are items in expect that not longer match anything but more specific items do (e.g. `microsoft` vs. `Microsoft`), there's now a specific category with help/advice
* Fancier excludes suggestions (excluding directories, file types, ...)
* Refreshed dictionaries
* The comment now links to the job summary (which includes SARIF link if available, the details view, and a generated commit that people can use if they're ok w/ the expect changes and don't want to run perl)



## Validation Steps Performed

1. the branch was developed in https://github.com/check-spelling-sandbox/terminal/actions?query=branch%3Acheck-spelling-0.0.22
2. ensuring compatibility with 0.0.21 was done in https://github.com/check-spelling-sandbox/terminal/pull/3
3. this version has been in development for a year and has quite a few improvements, we've been actively dogfooding it throughout this period 😄

## PR Checklist
   - If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
